File: hoststojson.py
#!/usr/bin/env python3

import sys
import os
import json

def generate_2d_list():
    input_file = sys.argv[1]

    dict_of_hosts = {}

    with open(input_file, "r") as f:
        list_of_2d = []

        for line in f:
            list_2d = []
            line = line.rstrip()

            if not "#" in line:
                word = line.split(" ")

                for domain in word[1:]:
                    if len(domain.rstrip().split()) > 0:
                        dict_of_hosts.update({ domain: word[0] })
        return dict_of_hosts


print(json.dumps(generate_2d_list(), indent=4))

File: hoststojson.js
#!/usr/bin/env node

const process = require('process');

function hostsToJson() {
    let userInput = process.argv[2]
    let data
    let hostsObject = {}

    if (!userInput == undefined || !userInput == "") {
        const fs = require('fs')

        try {
            data = fs.readFileSync(userInput, 'utf8')

            let lines = data.split("\n");

            lines.forEach((ele) => {
                if(!ele.includes("#")) {
                    let line = ele.split(" ")

                    if (!ele === "" || ele.length > 1) {
                        line.forEach((el) => {
                            if(el.length > 1) {

                                line.shift()

                                line.forEach((domain) => {
                                    if (!domain == "" || domain.length > 1) {
                                        if(domain != "localhost") {
                                            // I have to recreate the array here cuz I used the shift() function - this is cuz i need the ip back in this loop
                                            line = ele.split(" ")
                                            ip = line[0]
                                            hostsObject[`${domain.toString()}`] = ip;
                                        }
                                    }
                                })
                            }
                        })
                    }
                }
            })
            return hostsObject
        } catch (err) {
            console.log("Is arg1 really a file\n")
            process.exit(1)
        }
    } else {
        console.log("arg1 should be a hostfile\n./hoststojson <file>\n")
        process.exit(2)
    }
}


console.log(JSON.parse(JSON.stringify(hostsToJson())))